src/sqlfluff/rules/L064.py

```diff
@@ -0,0 +1,262 @@
+            memory = context.memory
+            preferred_string_quotes = memory.get("preferred_string_quotes")
+
+            if not preferred_string_quotes:
+                # Getting the quote from LAST character to be able to handle STRING
+                # prefixes
+                preferred_string_quotes = (
+                    "double_quotes"
+                    if context.segment.raw[-1] == '"'
+                    else "single_quotes"
+                )
+                memory["preferred_string_quotes"] = preferred_string_quotes
+                self.logger.debug(
+                    "Preferred string quotes is set to `consistent`. Derived quoting "
+                    "style %s from first quoted literal.",
+                    preferred_string_quotes,
+                )
+        else:
+            preferred_string_quotes = self.preferred_string_quotes
+
+        fixed_string = self._normalize_preferred_string_quotes(
+            context.segment.raw,
+            preferred_quote_char=QUOTES_MAPPING[preferred_string_quotes][
+                "preferred_quote_char"
+            ],
+            alternate_quote_char=QUOTES_MAPPING[preferred_string_quotes][
+                "alternate_quote_char"
+            ],
+        )
+
+        if fixed_string != context.segment.raw:
+            return LintResult(
+                anchor=context.segment,
+                memory=context.memory,
+                fixes=[
+                    LintFix.replace(
+                        context.segment,
+                        [
+                            CodeSegment(
+                                raw=fixed_string,
+                                name="quoted_literal",
+                                type="literal",
+                            )
+                        ],
+                    )
+                ],
+                description=(
+                    "Inconsistent use of preferred quote style '"
+                    f"{QUOTES_MAPPING[preferred_string_quotes]['common_name']}'. "
+                    f"Use {fixed_string} instead of {context.segment.raw}."
+                ),
+            )
+
+        return None
+
+    def _sub_twice(self, regex: regex.Pattern, replacement: str, original: str) -> str:
+        """Replace `regex` with `replacement` twice on `original`.
+
+        This is used by string normalization to perform replaces on overlapping matches.
+        """
+        return regex.sub(replacement, regex.sub(replacement, original))
+
+    def _normalize_preferred_string_quotes(
+        self, s: str, preferred_quote_char: str, alternate_quote_char: str
+    ) -> str:
+        """Prefer `preferred_quote_char` but only if it doesn't cause more escaping.
+
+        Adds or removes backslashes as appropriate.
+        """
+        value = s.lstrip(STRING_PREFIX_CHARS)
+
+        if value[:3] == preferred_quote_char * 3:
+            # TODO: Are we not replacing unnecessary quotes here?
```

Comment:
Do you mean unnecessary escaped quotes?

Reply:
Yeah, I meant unnecessary escaped quotes. After double checking more carefully I realised that the idea is to never replace escaped quotes in tripple quoted strings. I will update the comment.